aws/resource_aws_waf_rate_based_rule.go

```diff
@@ -143,6 +173,21 @@ func resourceAwsWafRateBasedRuleUpdate(d *schema.ResourceData, meta interface{})
 		}
 	}
 
+	arn := arn.ARN{
+		Partition: meta.(*AWSClient).partition,
+		Service:   "waf",
+		AccountID: meta.(*AWSClient).accountid,
+		Resource:  fmt.Sprintf("ratebasedrule/%s", d.Id()),
+	}.String()
+
+	if d.HasChange("tags") {
+		o, n := d.GetChange("tags")
+
+		if err := keyvaluetags.WafUpdateTags(conn, arn, o, n); err != nil {
```

Comment:
Since we are saving the `arn` attribute in the Terraform state, we can fetch it here instead of re-implementing the ARN logic 👍 

```suggestion
		if err := keyvaluetags.WafUpdateTags(conn, d.Get("arn").(string), o, n); err != nil {
```

and removing the `arn := arn.ARN{...}` bits right above this.

Reply:
I tried this but it doesnt work for me for some reason

```
=== RUN   TestAccAWSWafRateBasedRule_Tags
=== PAUSE TestAccAWSWafRateBasedRule_Tags
=== CONT  TestAccAWSWafRateBasedRule_Tags
--- FAIL: TestAccAWSWafRateBasedRule_Tags (36.52s)
    testing.go:569: Step 0 error: errors during apply:
        
        Error: error updating tags: error tagging resource (): InvalidParameter: 1 validation error(s) found.
        - minimum field size of 1, TagResourceInput.ResourceARN.
```

Reply:
fixed, the issue was that the create function called the update one and it messed with it.